src/params/params.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+var _ = require('underscore');
+var ad = require('../ad');
+var config = require('./config');
+var serializeParams = require('./serialize').serializeParams;
+
+
+// The local copy of the parameter table
+var _params;
+
+
+// Called before we start evaluating a webppl program.
+function init(k) {
+  var store = config.getStore();
+  if (!config.isManualId()) {
+    config.setFreshId();
+  }
+  return store.start(function() { return sync(k); });
+}
+
+
+function stop(k) {
+  var store = config.getStore();
+  return store.stop(k);
+}
+
+
+function sync(k) {
+  var store = config.getStore();
+  var next = function(params) {
+    if (!params) {
+      throw new Error('Expected store to return params, got', params);
+    }
+    _params = params;
+    return k(params);
+  };
+  return store.getParams(config.getId(), next);
+}
+
+
+// This is not a continuation-passing style function, since it doesn't
+// make use of any store functions that could be asynchronous. Instead,
+// it directly returns the current local parameter copy.
+function get() {
+  return _params;
+}
+
+
+// Save the local parameter table to a file
+function save(filename) {
+  var s = JSON.stringify(serializeParams(_params));
+  fs.writeFileSync(options.checkpointParamsFilename, s);
```

Comment:
`fs` is not required in this module, `options.checkpointParamsFilename` => `filename`?